c_glib/arrow-glib/basic-array.cpp

```diff
@@ -2161,6 +2161,27 @@ garrow_decimal128_array_format_value(GArrowDecimal128Array *array,
   return g_strndup(value.data(), value.size());
 }
 
+/**
+ * garrow_decimal128_array_get_value:
+ * @array: A #GArrowDecimal128Array.
+ * @i: The index of the target value.
+ *
+ * Returns: (transfer full): The i-th value.
+ *
+ * Since: 0.10.0
+ */
+GArrowDecimal128 *
+garrow_decimal128_array_get_value(GArrowDecimal128Array *array,
+                                  gint64 i)
+{
+  auto arrow_array = garrow_array_get_raw(GARROW_ARRAY(array));
+  auto arrow_decimal128_array =
+    static_cast<arrow::Decimal128Array *>(arrow_array.get());
```

Comment:
Ah, I missed it. We should use `std::static_pointer_cast<arrow::Decimal128Array>(arrow_array)` instead of `static_cast<... *>(...get())`.
Can you change the same code in `format_value()`?